lib/modules/userbarHider.js

```diff
@@ -102,8 +102,8 @@ function updateToggleButton(userbarHidden) {
 }
 
 function toggleUserbarElementsDisplay(userbarHidden) {
-	const elements = $(userbar).children().not($userbarToggle);
-
+	const elements = $(userbar).children().not($userbarToggle).not('.message-count');
```

Comment:
I don't think this will work properly when you have new messages; it won't hide the message-count icon.

I think the method described in #3128 will work better; that is use `BodyClasses.toggle(userbarHidden, 'res-hide-userbar');` here, and to the bottom of `_userbarHider.scss` add

```scss
.res-hide-userbar {
    #header-bottom-right > :not(#userbarToggle)
}
```

(or something along those lines)